### ISSUE TITLE
Vim-go cookbook install error on Ubuntu 14.04...

### ISSUE THREAD

**Issue by gregonplains** [mistsys/vim-go-cb#1]
I downloaded the vim-go cookbook and extracted the files to my cookbooks directory, and
I found when I executed the cookbook, there were messages requiring additional cookbooks
that should not have been required; over and above "apt git golang and vim" cookbooks.
The below session on my local system shows a listing of the additional cookbooks I downloaded, and
also shows the error message I received when executing the cookbook.  If there's any easier way
to execute "vim-go", please let me know.  Thanks. - Greg

gregory@gregory-VirtualBox:~/chef-repo$ cd cookbooks
gregory@gregory-VirtualBox:~/chef-repo/cookbooks$ ls -la
total 56
drwxrwxr-x 14 gregory gregory 4096 Nov  9 13:10 .
drwxrwxr-x  4 gregory gregory 4096 Nov  9 11:47 ..
drwxrwxr-x  9 gregory gregory 4096 Nov  9 12:53 apt
drwxrwxr-x  5 gregory gregory 4096 Nov  9 13:00 build-essential
drwxrwxr-x  8 gregory gregory 4096 Nov  9 13:08 chef_handler
drwxrwxr-x  7 gregory gregory 4096 Nov  9 13:03 dmg
drwxrwxr-x  8 gregory gregory 4096 Nov  9 12:55 git
drwxrwxr-x  8 gregory gregory 4096 Oct 19 12:05 golang
drwxrwxr-x  7 gregory gregory 4096 Nov  9 11:23 learn_chef_apache2
drwxrwxr-x  4 gregory gregory 4096 Nov  9 12:59 vim
drwxrwxr-x  6 gregory gregory 4096 Nov  9 12:41 vim-go
drwxrwxr-x  8 gregory gregory 4096 Nov  9 13:04 windows
drwxrwxr-x  8 gregory gregory 4096 Nov  9 13:10 yum
drwxrwxr-x  4 gregory gregory 4096 Nov  9 13:07 yum-epel
gregory@gregory-VirtualBox:~/chef-repo/cookbooks$ cd nodes
bash: cd: nodes: No such file or directory
gregory@gregory-VirtualBox:~/chef-repo/cookbooks$ ls -la ../nodes
total 72
drwxr-xr-x 2 root    root     4096 Nov  9 11:47 .
drwxrwxr-x 4 gregory gregory  4096 Nov  9 11:47 ..
-rw-r--r-- 1 root    root    64490 Nov  9 11:47 gregory-VirtualBox.json
gregory@gregory-VirtualBox:~/chef-repo/cookbooks$ sudo chef-client --local-mode --runlist 'recipe[vim-go]'
[sudo] password for gregory: 
[2015-11-09T13:25:17-09:00] WARN: No config file found or specified on command line, using command line options.
Starting Chef Client, version 12.5.1
resolving cookbooks for run list: ["vim-go"]
Synchronizing Cookbooks:
  - vim (2.0.0)
  - vim-go (0.1.2)
  - dmg (2.3.0)
  - apt (2.9.2)
  - build-essential (2.2.4)
  - git (4.3.4)
  - yum-epel (0.6.4)
  - windows (1.38.3)
  - chef_handler (1.2.0)
  - yum (3.8.2)
  - golang (1.7.0)
Compiling Cookbooks...

================================================================================
Recipe Compile Error in /home/gregory/.chef/local-mode-cache/cache/cookbooks/vim-go/recipes/default.rb
================================================================================

Net::HTTPServerException
------------------------
404 "Not Found"

Cookbook Trace:
---------------
  /home/gregory/.chef/local-mode-cache/cache/cookbooks/vim-go/recipes/default.rb:30:in `from_file'

Relevant File Content:
----------------------
/home/gregory/.chef/local-mode-cache/cache/cookbooks/vim-go/recipes/default.rb:

 23:  include_recipe "vim" if node[:vim_go][:install_vim]
 24:  
 25:  node.set['go']['version'] = "1.4.2"
 26:  node.set['go']['filename'] = "go#{node['go']['version']}.#{node['os']}-#{node['go']['platform']}.tar.gz"
 27:  node.set['go']['url'] = "http://golang.org/dl/#{node['go']['filename']}"
 28:  include_recipe 'golang'
 29:  
 30>> vim_users = search(:users, "vim:true")
 31:  
 32:  vim_users.each do |user_item|
 33:    user = user_item[:id]
 34:    Chef::Log.info "vim-go::default: Processing user #{user.inspect}"
 35:    home_dir = File.join(File.expand_path("~#{user}"))
 36:    next unless (user_item[:vim] && File.exist?(home_dir))
 37:    Chef::Log.info "vim-go::default: #{user}'s dir ${home_dir.inspect} exists and user_item[:vim] is true"
 38:    directory File.join(home_dir, ".vim") do
 39:      recursive true

Running handlers:
[2015-11-09T13:25:20-09:00] ERROR: Running exception handlers
Running handlers complete
[2015-11-09T13:25:20-09:00] ERROR: Exception handlers complete
Chef Client failed. 0 resources updated in 03 seconds
[2015-11-09T13:25:20-09:00] FATAL: Stacktrace dumped to /home/gregory/.chef/local-mode-cache/cache/chef-stacktrace.out
[2015-11-09T13:25:20-09:00] ERROR: 404 "Not Found"
[2015-11-09T13:25:21-09:00] FATAL: Chef::Exceptions::ChildConvergeError: Chef run process exited unsuccessfully (exit code 1)

**Comment by gregonplains** [comment]
I thought these dependency problems might be solved by using the chef server to deploy go-vim,
so I deleted all the cookbooks I downloaded, and instead, followed the instructions for LINUX on setting up the starter kit.  Looks like there are still some unanticipated dependences (other than apt, git, golang, and vim) to use vim-go.  You'll see in the below messages, git has some additional dependencies that aren't listed on the vim-go cookbook docs, specifically "windows", "yum-epel", and "dmg" which I don't think are applicable to ubuntu at all:

gregory@gregory-VirtualBox:~/chef-repo/cookbooks$ ls -la
total 24
drwxrwxr-x 4 gregory gregory 4096 Nov  9 14:14 .
drwxrwxr-x 6 gregory gregory 4096 Nov  9 13:50 ..
drwxrwxr-x 7 gregory gregory 4096 Nov  9 11:23 learn_chef_apache2
drwxr-xr-x 6 gregory gregory 4096 Nov  9  2015 starter
-rw------- 1 gregory gregory 4480 Nov  9 14:14 vim-go-0.1.2.tar.gz
gregory@gregory-VirtualBox:~/chef-repo/cookbooks$ rm vim*.gz
gregory@gregory-VirtualBox:~/chef-repo/cookbooks$ cd ..
gregory@gregory-VirtualBox:~/chef-repo$ knife cookbook site download vim-go
Downloading vim-go from Supermarket at version 0.1.2 to /home/gregory/chef-repo/vim-go-0.1.2.tar.gz
Cookbook saved: /home/gregory/chef-repo/vim-go-0.1.2.tar.gz
gregory@gregory-VirtualBox:~/chef-repo$ tar -zxvf vim-go-0.1.2.tar.gz -C cookbooks
vim-go/README.md
vim-go/CHANGELOG.md
vim-go/metadata.json
vim-go/attributes/default.rb
vim-go/files/default
vim-go/recipes/default.rb
vim-go/templates/default
vim-go/templates/default/README.md.erb
gregory@gregory-VirtualBox:~/chef-repo$ ls 
cookbooks  goodbye.rb  hello.rb  nodes  README.md  roles  vim-go-0.1.2.tar.gz  webserver.rb  webserverstop.rb
gregory@gregory-VirtualBox:~/chef-repo$ rm vim-go-0.1.2.tar.gz
gregory@gregory-VirtualBox:~/chef-repo$ knife cookbook upload vim-go
Uploading vim-go       [0.1.2]
ERROR: Cookbook vim-go depends on cookbooks which are not currently
ERROR: being uploaded and cannot be found on the server.
ERROR: The missing cookbook(s) are: 'apt' version '>= 0.0.0', 'git' version '>= 0.0.0', 'golang' version '>= 0.0.0', 'vim' version '>= 0.0.0'
gregory@gregory-VirtualBox:~/chef-repo$ knife cookbook site download apt
Downloading apt from Supermarket at version 2.9.2 to /home/gregory/chef-repo/apt-2.9.2.tar.gz
Cookbook saved: /home/gregory/chef-repo/apt-2.9.2.tar.gz
gregory@gregory-VirtualBox:~/chef-repo$ ls
apt-2.9.2.tar.gz  cookbooks  goodbye.rb  hello.rb  nodes  README.md  roles  webserver.rb  webserverstop.rb
gregory@gregory-VirtualBox:~/chef-repo$ tar -zxvf apt-2.9.2.tar.gz -C cookbooks
apt/README.md
apt/CHANGELOG.md
apt/metadata.json
apt/attributes/default.rb
apt/files/default
apt/files/default/15update-stamp
apt/files/default/apt-proxy-v2.conf
apt/libraries/helpers.rb
apt/libraries/matchers.rb
apt/libraries/network.rb
apt/providers/preference.rb
apt/providers/repository.rb
apt/recipes/cacher-client.rb
apt/recipes/cacher-ng.rb
apt/recipes/default.rb
apt/recipes/unattended-upgrades.rb
apt/resources/preference.rb
apt/resources/repository.rb
apt/templates/debian-6.0
apt/templates/debian-6.0/acng.conf.erb
apt/templates/default
apt/templates/default/01proxy.erb
apt/templates/default/10recommends.erb
apt/templates/default/20auto-upgrades.erb
apt/templates/default/50unattended-upgrades.erb
apt/templates/default/acng.conf.erb
apt/templates/default/unattended-upgrades.seed.erb
apt/templates/ubuntu-10.04
apt/templates/ubuntu-10.04/acng.conf.erb
gregory@gregory-VirtualBox:~/chef-repo$ rm apt-2.9.2.tar.gz
gregory@gregory-VirtualBox:~/chef-repo$ knife cookbook upload apt
Uploading apt          [2.9.2]
Uploaded 1 cookbook.
gregory@gregory-VirtualBox:~/chef-repo$ knife cookbook site download git
Downloading git from Supermarket at version 4.3.4 to /home/gregory/chef-repo/git-4.3.4.tar.gz
Cookbook saved: /home/gregory/chef-repo/git-4.3.4.tar.gz
gregory@gregory-VirtualBox:~/chef-repo$ tar -zxvf git-4.3.4.tar.gz -C cookbooks
git/README.md
git/CHANGELOG.md
git/metadata.json
git/attributes/default.rb
git/libraries/helpers.rb
git/libraries/matchers.rb
git/libraries/provider_git_client.rb
git/libraries/provider_git_client_osx.rb
git/libraries/provider_git_client_package.rb
git/libraries/provider_git_client_source.rb
git/libraries/provider_git_client_windows.rb
git/libraries/provider_git_service.rb
git/libraries/provider_git_service_xinetd.rb
git/libraries/resource_git_client.rb
git/libraries/resource_git_service.rb
git/libraries/z_provider_mapping.rb
git/providers/config.rb
git/recipes/default.rb
git/recipes/package.rb
git/recipes/server.rb
git/recipes/source.rb
git/recipes/windows.rb
git/resources/config.rb
git/templates/default
git/templates/default/git-xinetd.d.erb
git/templates/default/sv-git-daemon-log-run.erb
git/templates/default/sv-git-daemon-run.erb
gregory@gregory-VirtualBox:~/chef-repo$ ls
cookbooks  git-4.3.4.tar.gz  goodbye.rb  hello.rb  nodes  README.md  roles  webserver.rb  webserverstop.rb
gregory@gregory-VirtualBox:~/chef-repo$ rm git*
gregory@gregory-VirtualBox:~/chef-repo$ knife cookbook upload git
Uploading git            [4.3.4]
ERROR: Cookbook git depends on cookbooks which are not currently
ERROR: being uploaded and cannot be found on the server.
ERROR: The missing cookbook(s) are: 'build-essential' version '>= 0.0.0', 'dmg' version '>= 0.0.0', 'windows' version '>= 0.0.0', 'yum-epel' version '>= 0.0.0'

**Comment by gregonplains** [comment]
Forgive me, as this is my first time posting and didn't realise when I posted the 2nd time, the issue would be automatically closed.  Would still benefit from some assistance in installing this cookbook for vim-go.  Thanks. - Greg.

**Comment by rberger** [comment]
Hi Greg, Are you using Berkshelf to pull in all the dependencies?

If not, try creating a file `Berkshelf` in the top level of your chef-repo and put the following in it:

```
source "https://supermarket.chef.io"

metadata

cookbook 'vim-go', git: 'git@github.com:mistsys/vim-go-cb.git'
```

Then run:
```
berks update
berks upload
```

instead of using knife cookbook upload

**Comment by rberger** [comment]
You might have to make the Berksfile entry for `vim-go` use https instead of git.

```
cookbook 'vim-go', git: 'https://github.com/mistsys/vim-go-cb.git'
```
instead of
```
cookbook 'vim-go', git: 'git@github.com:mistsys/vim-go-cb.git'
```

**Comment by gregonplains** [comment]
Robert,

Thanks for getting back to me.

I managed to get vim-go installed using pathogen, but I really like the
idea of using the cookbooks, so I will follow
the advice you have here tomorrow.  Thanks. - Greg.

P.S.  Here's a King Salmon I pulled off the boat up here in Homer, AK last
month.  That's me standing with him.LOL.

**Comment by rberger** [comment]
Please send fresh Salmon :-)

**Comment by gregonplains** [comment]
Robert,

The boat is in the garage after today.  Too cold.

I did get rid of the vim-go cookbook I downloaded below, and
used berks to create a new vim-go cookbook, edit the Berksfile as per your
instructions above, then tried to
run update. Here's the result of that process:

gregory@gregory-VirtualBox:~/chef-repo/cookbooks$ rm -r vim-go
gregory@gregory-VirtualBox:~/chef-repo/cookbooks$ berks cookbook vim-go
      create  vim-go/files/default
      create  vim-go/templates/default
      create  vim-go/attributes
      create  vim-go/libraries
      create  vim-go/providers
      create  vim-go/recipes
      create  vim-go/resources
      create  vim-go/recipes/default.rb
      create  vim-go/metadata.rb
      create  vim-go/LICENSE
      create  vim-go/README.md
      create  vim-go/CHANGELOG.md
      create  vim-go/Berksfile
      create  vim-go/Thorfile
      create  vim-go/chefignore
      create  vim-go/.gitignore
      create  vim-go/Gemfile
      create  .kitchen.yml
    conflict  chefignore
Overwrite /home/gregory/chef-repo/cookbooks/vim-go/chefignore? (enter "h"
for help) [Ynaqdh] Y
       force  chefignore
      append  Thorfile
      create  test/integration/default
      append  .gitignore
      append  .gitignore
      append  Gemfile
      append  Gemfile
You must run `bundle install' to fetch any new gems.
      create  vim-go/Vagrantfile
gregory@gregory-VirtualBox:~/chef-repo/cookbooks$ ls -la
total 28
drwxrwxr-x  7 gregory gregory 4096 Nov 11 15:26 .
drwxrwxr-x  6 gregory gregory 4096 Nov 11 15:20 ..
drwxrwxr-x  9 gregory gregory 4096 Nov  9 14:20 apt
drwxrwxr-x  8 gregory gregory 4096 Nov  9 14:27 git
drwxr-xr-x  7 gregory gregory 4096 May  1  2015 learn_chef_apache2
drwxr-xr-x  6 gregory gregory 4096 Nov  9 22:59 starter
drwxrwxr-x 11 gregory gregory 4096 Nov 11 15:27 vim-go
gregory@gregory-VirtualBox:~/chef-repo/cookbooks$ cd vim-go
gregory@gregory-VirtualBox:~/chef-repo/cookbooks/vim-go$ ls -la
total 88
drwxrwxr-x 11 gregory gregory 4096 Nov 11 15:27 .
drwxrwxr-x  7 gregory gregory 4096 Nov 11 15:26 ..
drwxrwxr-x  2 gregory gregory 4096 Nov 11 15:26 attributes
-rw-rw-r--  1 gregory gregory   47 Nov 11 15:26 Berksfile
-rw-rw-r--  1 gregory gregory   35 Nov 11 15:26 CHANGELOG.md
-rw-rw-r--  1 gregory gregory    9 Nov 11 15:27 chefignore
drwxrwxr-x  3 gregory gregory 4096 Nov 11 15:26 files
-rw-rw-r--  1 gregory gregory  449 Nov 11 15:27 Gemfile
drwxrwxr-x  7 gregory gregory 4096 Nov 11 15:26 .git
-rw-rw-r--  1 gregory gregory  155 Nov 11 15:27 .gitignore
-rw-rw-r--  1 gregory gregory  173 Nov 11 15:26 .kitchen.yml
drwxrwxr-x  2 gregory gregory 4096 Nov 11 15:26 libraries
-rw-rw-r--  1 gregory gregory   72 Nov 11 15:26 LICENSE
-rw-rw-r--  1 gregory gregory  241 Nov 11 15:26 metadata.rb
drwxrwxr-x  2 gregory gregory 4096 Nov 11 15:26 providers
-rw-rw-r--  1 gregory gregory  598 Nov 11 15:26 README.md
drwxrwxr-x  2 gregory gregory 4096 Nov 11 15:26 recipes
drwxrwxr-x  2 gregory gregory 4096 Nov 11 15:26 resources
drwxrwxr-x  3 gregory gregory 4096 Nov 11 15:26 templates
drwxrwxr-x  3 gregory gregory 4096 Nov 11 15:27 test
-rw-rw-r--  1 gregory gregory  241 Nov 11 15:27 Thorfile
-rw-rw-r--  1 gregory gregory 3313 Nov 11 15:27 Vagrantfile
gregory@gregory-VirtualBox:~/chef-repo/cookbooks/vim-go$ vi Berksfile
gregory@gregory-VirtualBox:~/chef-repo/cookbooks/vim-go$ berks update
An error occurred while reading the Berksfile:

  Your Berksfile contains multiple entries named 'vim-go'. Please remove
duplicate dependencies, or put them in different groups.
gregory@gregory-VirtualBox:~/chef-repo/cookbooks/vim-go$ vi Berksfile
gregory@gregory-VirtualBox:~/chef-repo/cookbooks/vim-go$ cat Berksfile
source "https://supermarket.chef.io"

metadata
cookbook 'vim-go', git: 'https://github.com/mistsys/vim-go-cb.git'
gregory@gregory-VirtualBox:~/chef-repo/cookbooks/vim-go$ vi Berksfile
gregory@gregory-VirtualBox:~/chef-repo/cookbooks/vim-go$ berks update
An error occurred while reading the Berksfile:

  Your Berksfile contains multiple entries named 'vim-go'. Please remove
duplicate dependencies, or put them in different groups.
gregory@gregory-VirtualBox:~/chef-repo/cookbooks/vim-go$ cat Berksfile
source "https://supermarket.chef.io"

metadata
cookbook 'vim-go', git: 'http://github.com/mistsys/vim-go-cb.git'
(end session)

As you can see, when I run the "berks update", I get the same message
("Your Berksfile...") in either case, whether I use "http" or "https" with
the "cookbook"
command line I inserted into the Berksfile.

Any ideas? Thanks. - Greg.

**Comment by rberger** [comment]
So you don’t want to do
>  berks cookbook vim-go

That’s going to create a new “empty” cookbook call vim-go

You want to use the vim-go cookbook as a “library” cookbook. (unless you want to modify it but that’s a different story)

Do you already have a repo / cookbook, whatever set up to create the server that you want to install the vim-go stuff on?

If you have a cookbook, you should be able to just add vim-go as a dependency in the metadata of that cookbook and then say:

include_recipe ‘vim-go’

Keep the line
> cookbook 'vim-go', git: 'https://github.com/mistsys/vim-go-cb.git'

in your Berksfile

I’m not sure how into Chef you are already. So please don’t take this the wrong way, but you may want to check out:
https://learn.chef.io

to get better oriented on how to work with community cookbooks

Would love to help more but I’m really saturated at work right now.

Rob


> On Nov 11, 2015, at 4:40 PM, gregonplains <notifications@github.com> wrote:
> 
> Robert,
> 
> The boat is in the garage after today. Too cold.
> 
> I did get rid of the vim-go cookbook I downloaded below, and
> used berks to create a new vim-go cookbook, edit the Berksfile as per your
> instructions above, then tried to
> run update. Here's the result of that process:
> 
> gregory@gregory-VirtualBox:~/chef-repo/cookbooks$ rm -r vim-go
> gregory@gregory-VirtualBox:~/chef-repo/cookbooks$ berks cookbook vim-go
> create vim-go/files/default
> create vim-go/templates/default
> create vim-go/attributes
> create vim-go/libraries
> create vim-go/providers
> create vim-go/recipes
> create vim-go/resources
> create vim-go/recipes/default.rb
> create vim-go/metadata.rb
> create vim-go/LICENSE
> create vim-go/README.md
> create vim-go/CHANGELOG.md
> create vim-go/Berksfile
> create vim-go/Thorfile
> create vim-go/chefignore
> create vim-go/.gitignore
> create vim-go/Gemfile
> create .kitchen.yml
> conflict chefignore
> Overwrite /home/gregory/chef-repo/cookbooks/vim-go/chefignore? (enter "h"
> for help) [Ynaqdh] Y
> force chefignore
> append Thorfile
> create test/integration/default
> append .gitignore
> append .gitignore
> append Gemfile
> append Gemfile
> You must run `bundle install' to fetch any new gems.
> create vim-go/Vagrantfile
> gregory@gregory-VirtualBox:~/chef-repo/cookbooks$ ls -la
> total 28
> drwxrwxr-x 7 gregory gregory 4096 Nov 11 15:26 .
> drwxrwxr-x 6 gregory gregory 4096 Nov 11 15:20 ..
> drwxrwxr-x 9 gregory gregory 4096 Nov 9 14:20 apt
> drwxrwxr-x 8 gregory gregory 4096 Nov 9 14:27 git
> drwxr-xr-x 7 gregory gregory 4096 May 1 2015 learn_chef_apache2
> drwxr-xr-x 6 gregory gregory 4096 Nov 9 22:59 starter
> drwxrwxr-x 11 gregory gregory 4096 Nov 11 15:27 vim-go
> gregory@gregory-VirtualBox:~/chef-repo/cookbooks$ cd vim-go
> gregory@gregory-VirtualBox:~/chef-repo/cookbooks/vim-go$ ls -la
> total 88
> drwxrwxr-x 11 gregory gregory 4096 Nov 11 15:27 .
> drwxrwxr-x 7 gregory gregory 4096 Nov 11 15:26 ..
> drwxrwxr-x 2 gregory gregory 4096 Nov 11 15:26 attributes
> -rw-rw-r-- 1 gregory gregory 47 Nov 11 15:26 Berksfile
> -rw-rw-r-- 1 gregory gregory 35 Nov 11 15:26 CHANGELOG.md
> -rw-rw-r-- 1 gregory gregory 9 Nov 11 15:27 chefignore
> drwxrwxr-x 3 gregory gregory 4096 Nov 11 15:26 files
> -rw-rw-r-- 1 gregory gregory 449 Nov 11 15:27 Gemfile
> drwxrwxr-x 7 gregory gregory 4096 Nov 11 15:26 .git
> -rw-rw-r-- 1 gregory gregory 155 Nov 11 15:27 .gitignore
> -rw-rw-r-- 1 gregory gregory 173 Nov 11 15:26 .kitchen.yml
> drwxrwxr-x 2 gregory gregory 4096 Nov 11 15:26 libraries
> -rw-rw-r-- 1 gregory gregory 72 Nov 11 15:26 LICENSE
> -rw-rw-r-- 1 gregory gregory 241 Nov 11 15:26 metadata.rb
> drwxrwxr-x 2 gregory gregory 4096 Nov 11 15:26 providers
> -rw-rw-r-- 1 gregory gregory 598 Nov 11 15:26 README.md
> drwxrwxr-x 2 gregory gregory 4096 Nov 11 15:26 recipes
> drwxrwxr-x 2 gregory gregory 4096 Nov 11 15:26 resources
> drwxrwxr-x 3 gregory gregory 4096 Nov 11 15:26 templates
> drwxrwxr-x 3 gregory gregory 4096 Nov 11 15:27 test
> -rw-rw-r-- 1 gregory gregory 241 Nov 11 15:27 Thorfile
> -rw-rw-r-- 1 gregory gregory 3313 Nov 11 15:27 Vagrantfile
> gregory@gregory-VirtualBox:~/chef-repo/cookbooks/vim-go$ vi Berksfile
> gregory@gregory-VirtualBox:~/chef-repo/cookbooks/vim-go$ berks update
> An error occurred while reading the Berksfile:
> 
> Your Berksfile contains multiple entries named 'vim-go'. Please remove
> duplicate dependencies, or put them in different groups.
> gregory@gregory-VirtualBox:~/chef-repo/cookbooks/vim-go$ vi Berksfile
> gregory@gregory-VirtualBox:~/chef-repo/cookbooks/vim-go$ cat Berksfile
> source "https://supermarket.chef.io"
> 
> metadata
> cookbook 'vim-go', git: 'https://github.com/mistsys/vim-go-cb.git'
> gregory@gregory-VirtualBox:~/chef-repo/cookbooks/vim-go$ vi Berksfile
> gregory@gregory-VirtualBox:~/chef-repo/cookbooks/vim-go$ berks update
> An error occurred while reading the Berksfile:
> 
> Your Berksfile contains multiple entries named 'vim-go'. Please remove
> duplicate dependencies, or put them in different groups.
> gregory@gregory-VirtualBox:~/chef-repo/cookbooks/vim-go$ cat Berksfile
> source "https://supermarket.chef.io"
> 
> metadata
> cookbook 'vim-go', git: 'http://github.com/mistsys/vim-go-cb.git'
> (end session)
> 
> As you can see, when I run the "berks update", I get the same message
> ("Your Berksfile...") in either case, whether I use "http" or "https" with
> the "cookbook"
> command line I inserted into the Berksfile.
> 
> Any ideas? Thanks. - Greg.
> 
> On Mon, Nov 9, 2015 at 5:24 PM, Robert J. Berger <notifications@github.com>
> wrote:
> 
> > Please send fresh Salmon :-)
> >
> > —
> > Reply to this email directly or view it on GitHub
> > <https://github.com/mistsys/vim-go-cb/issues/1#issuecomment-155262898>.
> >
> —
> Reply to this email directly or view it on GitHub <https://github.com/mistsys/vim-go-cb/issues/1#issuecomment-155958323>.
> 

__________________________________
Robert J. Berger - Cloud Architect
Mist Systems, Inc.
rberger@mist.com - +1 408-838-8896

**Comment by gregonplains** [comment]
Robert,

This looked like it worked: 

gregory@gregory-VirtualBox:~/chef-repo/cookbooks/vim-go$ berks update
Lockfile not found! Run `berks install` to create the lockfile.
gregory@gregory-VirtualBox:~/chef-repo/cookbooks/vim-go$ berks install
Resolving cookbook dependencies...
Fetching 'vim-go' from source at .
Fetching cookbook index from https://supermarket.chef.io...
Installing apt (2.9.2)
Installing build-essential (2.2.4)
Installing chef_handler (1.2.0)
Installing dmg (2.3.0)
Installing git (4.3.4)
Installing golang (1.7.0)
Installing vim (2.0.0)
Using vim-go (0.1.0) from source at .
Installing windows (1.38.3)
Installing yum (3.8.2)
Installing yum-epel (0.6.4)
gregory@gregory-VirtualBox:~/chef-repo/cookbooks/vim-go$ cat Berksfile
source "https://supermarket.chef.io"

metadata

cookbook "apt" 
cookbook "git" 
cookbook "golang" 
cookbook "vim" 
gregory@gregory-VirtualBox:~/chef-repo/cookbooks/vim-go$ berks update
Resolving cookbook dependencies...
Fetching 'vim-go' from source at .
Fetching cookbook index from https://supermarket.chef.io...
Using apt (2.9.2)
Using build-essential (2.2.4)
Using chef_handler (1.2.0)
Using dmg (2.3.0)
Using golang (1.7.0)
Using git (4.3.4)
Using vim (2.0.0)
Using vim-go (0.1.0) from source at .
Using windows (1.38.3)
Using yum (3.8.2)
Using yum-epel (0.6.4)
gregory@gregory-VirtualBox:~/chef-repo/cookbooks/vim-go$ berks upload
Uploaded apt (2.9.2) to: 'https://api.chef.io:443/organizations/gregonplains'
Uploaded build-essential (2.2.4) to: 'https://api.chef.io:443/organizations/gregonplains'
Uploaded chef_handler (1.2.0) to: 'https://api.chef.io:443/organizations/gregonplains'
Uploaded dmg (2.3.0) to: 'https://api.chef.io:443/organizations/gregonplains'
Uploaded git (4.3.4) to: 'https://api.chef.io:443/organizations/gregonplains'
Uploaded golang (1.7.0) to: 'https://api.chef.io:443/organizations/gregonplains'
Uploaded vim (2.0.0) to: 'https://api.chef.io:443/organizations/gregonplains'
Uploaded vim-go (0.1.0) to: 'https://api.chef.io:443/organizations/gregonplains'
Uploaded windows (1.38.3) to: 'https://api.chef.io:443/organizations/gregonplains'
Uploaded yum (3.8.2) to: 'https://api.chef.io:443/organizations/gregonplains'
Uploaded yum-epel (0.6.4) to: 'https://api.chef.io:443/organizations/gregonplains'

**Comment by mist-slackbot** [comment]
Were you able to deploy it to your server?

> On Nov 11, 2015, at 5:26 PM, gregonplains <notifications@github.com> wrote:
> 
> Robert,
> 
> This looked like it worked:
> 
> gregory@gregory-VirtualBox:~/chef-repo/cookbooks/vim-go$ berks update
> Lockfile not found! Run berks install to create the lockfile.
> gregory@gregory-VirtualBox:~/chef-repo/cookbooks/vim-go$ berks install
> Resolving cookbook dependencies...
> Fetching 'vim-go' from source at .
> Fetching cookbook index from https://supermarket.chef.io <https://supermarket.chef.io/>...
> Installing apt (2.9.2)
> Installing build-essential (2.2.4)
> Installing chef_handler (1.2.0)
> Installing dmg (2.3.0)
> Installing git (4.3.4)
> Installing golang (1.7.0)
> Installing vim (2.0.0)
> Using vim-go (0.1.0) from source at .
> Installing windows (1.38.3)
> Installing yum (3.8.2)
> Installing yum-epel (0.6.4)
> gregory@gregory-VirtualBox:~/chef-repo/cookbooks/vim-go$ cat Berksfile
> source "https://supermarket.chef.io <https://supermarket.chef.io/>"
> 
> metadata
> 
> cookbook "apt" 
> cookbook "git" 
> cookbook "golang" 
> cookbook "vim" 
> gregory@gregory-VirtualBox:~/chef-repo/cookbooks/vim-go$ berks update
> Resolving cookbook dependencies...
> Fetching 'vim-go' from source at .
> Fetching cookbook index from https://supermarket.chef.io <https://supermarket.chef.io/>...
> Using apt (2.9.2)
> Using build-essential (2.2.4)
> Using chef_handler (1.2.0)
> Using dmg (2.3.0)
> Using golang (1.7.0)
> Using git (4.3.4)
> Using vim (2.0.0)
> Using vim-go (0.1.0) from source at .
> Using windows (1.38.3)
> Using yum (3.8.2)
> Using yum-epel (0.6.4)
> gregory@gregory-VirtualBox:~/chef-repo/cookbooks/vim-go$ berks upload
> Uploaded apt (2.9.2) to: 'https://api.chef.io:443/organizations/gregonplains <https://api.chef.io/organizations/gregonplains>'
> Uploaded build-essential (2.2.4) to: 'https://api.chef.io:443/organizations/gregonplains <https://api.chef.io/organizations/gregonplains>'
> Uploaded chef_handler (1.2.0) to: 'https://api.chef.io:443/organizations/gregonplains <https://api.chef.io/organizations/gregonplains>'
> Uploaded dmg (2.3.0) to: 'https://api.chef.io:443/organizations/gregonplains <https://api.chef.io/organizations/gregonplains>'
> Uploaded git (4.3.4) to: 'https://api.chef.io:443/organizations/gregonplains <https://api.chef.io/organizations/gregonplains>'
> Uploaded golang (1.7.0) to: 'https://api.chef.io:443/organizations/gregonplains <https://api.chef.io/organizations/gregonplains>'
> Uploaded vim (2.0.0) to: 'https://api.chef.io:443/organizations/gregonplains <https://api.chef.io/organizations/gregonplains>'
> Uploaded vim-go (0.1.0) to: 'https://api.chef.io:443/organizations/gregonplains <https://api.chef.io/organizations/gregonplains>'
> Uploaded windows (1.38.3) to: 'https://api.chef.io:443/organizations/gregonplains <https://api.chef.io/organizations/gregonplains>'
> Uploaded yum (3.8.2) to: 'https://api.chef.io:443/organizations/gregonplains <https://api.chef.io/organizations/gregonplains>'
> Uploaded yum-epel (0.6.4) to: 'https://api.chef.io:443/organizations/gregonplains <https://api.chef.io/organizations/gregonplains>'
> 
> —
> Reply to this email directly or view it on GitHub <https://github.com/mistsys/vim-go-cb/issues/1#issuecomment-155967918>.
> 

__________________________________
Robert J. Berger - Cloud Architect
Mist Systems, Inc.
rberger@mist.com - +1 408-838-8896

**Comment by gregonplains** [comment]
Yes.  Taking Robert's advice, and following along in the online tutorial, so I used the virtual machine you guys made available, then the instructions for the knife command:

knife bootstrap ADDRESS --ssh-user USER --ssh-password 'PASSWORD' --sudo --use-sudo-password --node-name node1 --run-list 'recipe[learn_chef_apache2]'

Of course, I substituted 'vim-go' for the recipe, and used the information for the virtual's username and password supplied from the instructions.

I'm getting the hang of things.

Thanks for your interest. - Greg.
